parquet-variant/src/main/java/org/apache/parquet/variant/VariantColumnConverter.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.variant;
+
+import static org.apache.parquet.schema.PrimitiveType.PrimitiveTypeName.BINARY;
+
+import java.util.List;
+import org.apache.parquet.io.api.Binary;
+import org.apache.parquet.schema.GroupType;
+import org.apache.parquet.schema.Type;
+
+/**
+ * Converter for shredded Variant values. Connectors should implement the addVariant method, similar to
+ * the add* methods on PrimitiveConverter.
+ */
+public abstract class VariantColumnConverter extends VariantElementConverter {
+
+  private int topLevelMetadataIdx = -1;
+
+  public VariantColumnConverter(GroupType variantSchema) {
+    super(variantSchema);
+
+    List<Type> fields = variantSchema.getFields();
+    for (int i = 0; i < fields.size(); i++) {
+      Type field = fields.get(i);
+      String fieldName = field.getName();
+      if (fieldName.equals("metadata")) {
+        this.topLevelMetadataIdx = i;
+        if (!field.isPrimitive() || field.asPrimitiveType().getPrimitiveTypeName() != BINARY) {
+          throw new IllegalArgumentException("Metadata must be a binary value");
+        }
+      }
+    }
+    if (topLevelMetadataIdx < 0) {
+      throw new IllegalArgumentException("Metadata missing from schema");
+    }
```

Comment:
It looks like the loop and the check above this line could be replaced with:

```java
    this.topLevelMetadataIdx = variantSchema.getFieldIndex("metadata");
```

That will throw an exception if `metadata` isn't found.